nUpdate.Administration/UI/Dialogs/PackageAddDialog.cs

```diff
@@ -1508,6 +1519,113 @@ private void infoButton_Click(object sender, EventArgs e)
+                        requiredAssemblyRootPathComboBox.Text + Char.ConvertFromUtf32(92) + requiredAssemblyPathTextBox.Text,
+                        new Version((int)requiredAssemblyMajorNumericUpDown.Value, (int)requiredAssemblyMinorNumericUpDown.Value, (int)requiredAssemblyBuildNumericUpDown.Value, (int)requiredAssemblyRevisionNumericUpDown.Value),
+                        null);
+                    }
+                    break;
+                case 3:
+                    if (String.IsNullOrWhiteSpace(requiredRegistryKeyPathTextBox.Text))
+                    {
+                        Popup.ShowPopup(SystemIcons.Exclamation, "No registry path set", new Exception("The registry key path is not valid!"), PopupButtons.Ok);
+                    }else if (String.IsNullOrWhiteSpace(requiredRegistryKeyValueTextBox.Text))
+                    {
+                        Popup.ShowPopup(SystemIcons.Exclamation, "No registry value set", new Exception("The registry value is not valid!"), PopupButtons.Ok);
+                    }
+                    else
+                    {
+                        _updateRequirement = new UpdateRequirement(
+                        UpdateRequirement.RequirementType.registry,
+                        "HKEY_" + requiredRegistryRootPathComboBox.Text + "\\" + requiredRegistryKeyPathTextBox.Text,
+                        null,
+                        requiredRegistryKeyValueTextBox.Text);
+                    }
+                    break;
+                default:
+                    _updateRequirement = null;
+                    break;
+            }
+            if(_updateRequirement != null)
+            {
+                _updateRequirements.Add(_updateRequirement);
+                requirementsListBox.Items.Add(_updateRequirement);
+            }
+        }
+
+        private Version osVersion(string osVersion)
+        {
+            osVersion = osVersion.Replace("Windows ", "");
+            switch (osVersion)
+            {
+                case "Vista":
+                    return new Version("6.0.6000.0");
+                case "Vista Service Pack 1":
+                    return new Version("6.0.6001.0");
+                case "Vista Service Pack 2":
+                    return new Version("6.0.6002.0");
+                case "7":
+                    return new Version("6.1.7600.0");
+                case "7 Service Pack 1":
+                    return new Version("6.1.7601.0");
+                case "8":
+                    return new Version("6.2.9200.0");
+                case "8.1":
+                    return new Version("6.3.9600.0");
+                case "10":
+                    return new Version("10.0.0.0");
+                default:
+                    return null;
+            }
+        }
+
+        private void requiredAssemblyPathButton_Click(object sender, EventArgs e)
+        {
+            OpenFileDialog dialog = new OpenFileDialog();
```

Comment:
Please dispose the OpenFileDialog or use a "using"-block.
Also make sure that the filter's file descriptions have capital letters and that you don't allow "All files". If the assembly part will be excluded due to my comment on the whole PullRequest, you can ignore this.